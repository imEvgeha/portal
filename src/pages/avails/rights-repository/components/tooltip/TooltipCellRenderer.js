import React, {useState} from 'react';
import PropTypes from 'prop-types';
import Button from '@atlaskit/button';
import EditorMediaWrapLeftIcon from '@atlaskit/icon/glyph/editor/media-wrap-left';
import classnames from 'classnames';
import {useDateTimeContext} from '../../../../../ui/elements/nexus-date-time-context/NexusDateTimeProvider';
import CustomActionsCellRenderer from '../../../../../ui/elements/nexus-grid/elements/cell-renderer/CustomActionsCellRenderer';
import {getDomainName} from '../../../../../util/Common';
import {DATETIME_FIELDS} from '../../../../../util/date-time/constants';
import {calculateIndicatorType, INDICATOR_SUCCESS, INDICATOR_RED} from '../../util/indicator';
import {ERROR_NO_CORE_TITLE_ID, FIND_MATCH, MATCHED_TITLE, NO_MATCHING_TITLE, VIEW_TITLE} from './constants';
import './TooltipCellRenderer.scss';

const TooltipCellRenderer = ({data = {}, isTooltipEnabled, setSingleRightMatch}) => {
    const [isTitleMatchTooltipOpen, setIsTitleMatchTooltipOpen] = useState(false);
    const {id} = data || {};
    const indicator = calculateIndicatorType(data);
    const notificationClass = indicator !== INDICATOR_RED ? '--success' : '--error';
    const toggleTooltip = () => setIsTitleMatchTooltipOpen(!isTitleMatchTooltipOpen);
    const {renderDateTime} = useDateTimeContext();

    const renderContent = () => {
        switch (calculateIndicatorType(data)) {
            case INDICATOR_RED:
                return (
                    <span>
                        {NO_MATCHING_TITLE}
                        <Button appearance="link" onClick={() => setSingleRightMatch([data])}>
                            {FIND_MATCH}
                        </Button>
                    </span>
                );
            case INDICATOR_SUCCESS:
                return (
                    <span>
                        {`${MATCHED_TITLE}  ${renderDateTime(
                            data.lastUpdateReceivedAt,
                            DATETIME_FIELDS.BUSINESS_DATETIME,
                            true,
                            true
                        )}`}
                        <Button
                            appearance="link"
                            onClick={() =>
                                window.open(`${getDomainName()}/metadata/detail/${data.coreTitleId}`, '_blank')
                            }
                        >
                            {VIEW_TITLE}
                        </Button>
                    </span>
                );
            default:
                return <span>{ERROR_NO_CORE_TITLE_ID}</span>;
        }
    };

    return (
        <CustomActionsCellRenderer id={id}>
            <div onClick={isTooltipEnabled ? toggleTooltip : null}>
                <EditorMediaWrapLeftIcon />
                <span
                    className={classnames(
                        'nexus-c-right-to-match-view__buttons_notification',
                        `nexus-c-right-to-match-view__buttons_notification${notificationClass}`
                    )}
                />
                {isTitleMatchTooltipOpen && (
                    <div className="nexus-c-tooltip-cell-renderer__tooltip">{renderContent()}</div>
                )}
            </div>
        </CustomActionsCellRenderer>
    );
};

TooltipCellRenderer.propTypes = {
    data: PropTypes.object,
    isTooltipEnabled: PropTypes.bool,
    setSingleRightMatch: PropTypes.func,
};

TooltipCellRenderer.defaultProps = {
    data: {},
    isTooltipEnabled: true,
    setSingleRightMatch: () => null,
};

export default TooltipCellRenderer;
