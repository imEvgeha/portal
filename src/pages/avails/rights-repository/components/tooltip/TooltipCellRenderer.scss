.nexus-c-tooltip-cell-renderer__tooltip {
    display: flex;
    justify-content: center;
    align-items: center;

    position: absolute;
    top: 2rem;
    height: 2rem;
    padding: 5px 15px;

    background: white;
    border: 1px solid black;
    border-radius: 3px;

    span {
        display: flex;
        align-items: flex-end;
        justify-content: center;
    }
}

.nexus-c-right-to-match-view__button-links {
    .p-button-label {
        font-size: 12px;
    }
}
