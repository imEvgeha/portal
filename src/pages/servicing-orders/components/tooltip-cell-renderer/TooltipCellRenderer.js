import React, {useState} from 'react';
import PropTypes from 'prop-types';
import Popup from '@atlaskit/popup';
import StatusTag from '@vubiquity-nexus/portal-ui/lib/elements/nexus-status-tag/StatusTag';
import ServicingOrdersTableStatusTooltip from '../servicing-orders-table-status-tooltip/ServicingOrdersTableStatusTooltip';

const TooltipCellRenderer = props => {
    const {value, soNumber} = props;
    const [isOpen, setIsOpen] = useState(false);
    return (
        <Popup
            isOpen={isOpen}
            onClose={() => setIsOpen(false)}
            boundariesElement="scrollParent"
            content={() => <ServicingOrdersTableStatusTooltip soNumber={soNumber} />}
            trigger={triggerProps => (
                <div {...triggerProps} onClick={() => setIsOpen(!isOpen)} role="button">
                    <StatusTag status={`FO_${value}`} />
                </div>
            )}
        />
    );
};

TooltipCellRenderer.propTypes = {
    value: PropTypes.string,
    soNumber: PropTypes.string,
};

TooltipCellRenderer.defaultProps = {
    value: '',
    soNumber: '',
};

export default TooltipCellRenderer;
