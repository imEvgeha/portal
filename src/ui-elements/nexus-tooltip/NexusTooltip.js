import React from 'react';
import Tooltip from '@atlaskit/tooltip';

function NexusTooltip({content, children, ...restProps}) {
    return (
        <Tooltip
            content={content}
            {...restProps}
        >
            {children}
        </Tooltip>
    );
}

NexusTooltip.defaultProps = {
    content: null,
};

export default NexusTooltip;