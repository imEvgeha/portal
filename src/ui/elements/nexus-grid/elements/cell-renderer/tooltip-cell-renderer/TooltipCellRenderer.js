import React, {useState} from 'react';
import PropTypes from 'prop-types';
import Popup from '@atlaskit/popup';
import ServicingOrdersTableStatusTooltip from '../../../../../../pages/servicing-orders/components/servicing-orders-table-status-tooltip/ServicingOrdersTableStatusTooltip';
import StatusTag from '../../../../nexus-status-tag/StatusTag';

const TooltipCellRenderer = data => {
    const {value, soNumber} = data;
    const [isOpen, setIsOpen] = useState(false);
    return (
        <Popup
            isOpen={isOpen}
            onClose={() => setIsOpen(false)}
            boundariesElement="scrollParent"
            content={() => <ServicingOrdersTableStatusTooltip soNumber={soNumber} />}
            trigger={triggerProps => (
                <div {...triggerProps} onClick={() => setIsOpen(!isOpen)} role="button">
                    <StatusTag status={`FO_${value}`} />
                </div>
            )}
        />
    );
};

TooltipCellRenderer.propTypes = {
    value: PropTypes.string,
};

TooltipCellRenderer.defaultProps = {
    value: '',
};

export default TooltipCellRenderer;
