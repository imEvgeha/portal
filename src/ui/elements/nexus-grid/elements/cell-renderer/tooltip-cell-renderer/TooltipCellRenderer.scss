.nexus-c-tooltip-cell-renderer {

}
