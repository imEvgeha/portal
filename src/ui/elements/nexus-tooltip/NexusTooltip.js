import React, {Fragment} from 'react';
import PropTypes from 'prop-types';
import Tooltip from '@atlaskit/tooltip';

const NexusTooltip = ({content, isDisabled, children, ...restProps}) => {
    return (
        <Tooltip
            content={!isDisabled && content}
            {...restProps}
        >
            <>
                {children}
            </>
        </Tooltip>
    );
};

NexusTooltip.propTypes = {
    content: PropTypes.node,
    isDisabled: PropTypes.bool,
};

NexusTooltip.defaultProps = {
    content: null,
    isDisabled: false,
};

export default NexusTooltip;
