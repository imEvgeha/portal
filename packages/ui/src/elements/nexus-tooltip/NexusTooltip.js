import React from 'react';
import PropTypes from 'prop-types';
import Tooltip from '@atlaskit/tooltip';
import {TooltipPrimitive} from '@atlaskit/tooltip/styled';
import styled from 'styled-components';

const NexusTooltip = ({content, isDisabled, children, hasWhiteBackground = false, ...restProps}) => {
    const InlineDialog = styled(TooltipPrimitive)`
        background: white;
        border-radius: 4px;
        box-shadow: 0 1px 2px rgba(0, 0, 0, 0.2);
        box-sizing: content-box;
        color: #333;
        padding: 8px 12px;
    `;

    const additionalProps = hasWhiteBackground ? {...restProps, component: InlineDialog} : restProps;
    return (
        <Tooltip content={!isDisabled && content} {...additionalProps}>
            <>{children}</>
        </Tooltip>
    );
};

NexusTooltip.propTypes = {
    content: PropTypes.node,
    isDisabled: PropTypes.bool,
    hasWhiteBackground: PropTypes.bool,
};

NexusTooltip.defaultProps = {
    content: null,
    isDisabled: false,
    hasWhiteBackground: false,
};

export default NexusTooltip;
