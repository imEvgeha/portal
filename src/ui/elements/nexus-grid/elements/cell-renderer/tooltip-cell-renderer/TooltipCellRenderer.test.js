import React from 'react';
import {shallow} from 'enzyme';
import NexusTooltip from "../../../../nexus-tooltip/NexusTooltip";
import TooltipCellRenderer from './TooltipCellRenderer';

describe('TooltipCellRenderer', () => {
    let wrapper = null;

    beforeEach(() => {
        wrapper = shallow(<TooltipCellRenderer value="12" />);
    });

    it('should match snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render a NexusTooltip', () => {
        const tooltip = wrapper.find(NexusTooltip);
        expect(tooltip.exists()).toBeTruthy();
        expect(tooltip.props().hasWhiteBackground).toEqual(true);

    });
});
