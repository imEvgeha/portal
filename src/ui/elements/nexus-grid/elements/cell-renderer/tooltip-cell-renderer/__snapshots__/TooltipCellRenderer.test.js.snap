// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TooltipCellRenderer should match snapshot 1`] = `
<Memo()
  boundariesElement="scrollParent"
  content={[Function]}
  isOpen={false}
  onClose={[Function]}
  trigger={[Function]}
/>
`;
